pkg/configuration/configuration.go

```diff
@@ -79,18 +78,9 @@ func GetRegistrationServiceConfig() RegistrationServiceConfig {
 	return NewRegistrationServiceConfig(config, secrets)
 }
 
-// ForceLoadRegistrationServiceConfig updates the cache using the provided client and returns the latest RegistrationServiceConfig
-func ForceLoadRegistrationServiceConfig(cl client.Client) (RegistrationServiceConfig, error) {
-	if configurationClient == nil {
-		configurationClient = cl
-	}
-	config, secrets, err := commonconfig.LoadLatest(cl, &toolchainv1alpha1.ToolchainConfig{})
-	if err != nil {
-		// return default config
-		logger.Error(err, "failed to force load RegistrationServiceConfig")
-		return RegistrationServiceConfig{cfg: &toolchainv1alpha1.ToolchainConfigSpec{}}, err
-	}
-	return NewRegistrationServiceConfig(config, secrets), nil
+// SetClient sets the client to be used to fetch the configuration
+func SetClient(cl client.Client) {
+	configurationClient = cl
```

Comment:
This looks like Java setter, I know. It's there to make the current structure and test suites happy. As soon as we use the same cached client everywhere, then we will be able to drop this part.